Content/Items/Misc/Accessories.AxeBook.cs

```diff
@@ -0,0 +1,541 @@
+﻿using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Core.Systems.CameraSystem;
+using StarlightRiver.Helpers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.GameContent;
+using Terraria.Graphics.Effects;
+
+namespace StarlightRiver.Content.Items.Misc
+{
+	internal class AxeBook : SmartAccessory
+	{
+		public int comboState;
+
+		public override string Texture => AssetDirectory.MiscItem + "AxeBook";
+
+		public AxeBook() : base("Tiger Technique", "Allows execution of combos with axes\nThe final strike will rend enemy armor\nRight click to throw your axe") { }
+
+		public override void Load()
+		{
+			StarlightItem.CanUseItemEvent += OverrideAxeEffects;
+			StarlightItem.AltFunctionUseEvent += AllowRightClick;
+		}
+
+		public override void Unload()
+		{
+			StarlightItem.CanUseItemEvent -= OverrideAxeEffects;
+			StarlightItem.AltFunctionUseEvent -= AllowRightClick;
+		}
+
+		public override void SafeSetDefaults()
+		{
+			Item.rare = Terraria.ID.ItemRarityID.Orange;
+		}
+
+		/// <summary>
+		/// Allows the player to right click with axes that dont normally have them
+		/// </summary>
+		/// <param name="item"></param>
+		/// <param name="player"></param>
+		/// <returns></returns>
+		private bool AllowRightClick(Item item, Player player)
+		{
+			if (Equipped(player))
+			{
+				if (item.CountsAsClass(DamageClass.Melee) && item.pick <= 0 && item.axe > 0 && item.hammer <= 0 && item.shoot <= 0 && item.useStyle == Terraria.ID.ItemUseStyleID.Swing && !item.noMelee)
+					return true;
+			}
+
+			return false;
+		}
+
+		/// <summary>
+		/// Changes the effect of using an item with axe power, melee damage, no projectile, and a swing usestyle to the intended alternative effect
+		/// </summary>
+		/// <param name="item"></param>
+		/// <param name="player"></param>
+		/// <returns></returns>
+		private bool OverrideAxeEffects(Item item, Player player)
```

Comment:
This probably shouldn't entirely override axe functionality. I would expect to still be able to chop trees down with this equipped